client/api_client.go

```diff
@@ -208,6 +232,27 @@ func New(chainID string, endpoint string, option Option) (IClient, error) {
 		}
 	}
 
+	// register off-chain-auth-v2 pubkey to all sps
+	if option.OffChainAuthOptionV2 != nil {
+		if option.OffChainAuthOptionV2.Seed == "" || option.OffChainAuthOptionV2.Domain == "" {
+			return nil, errors.New("seed and domain can't be empty in OffChainAuthOptionV2")
+		}
+		_, userEddsaPublicKey := GetEd25519PrivateKeyAndPublicKey(option.OffChainAuthOptionV2.Seed)
+		option.OffChainAuthOptionV2.PublicKey = hex.EncodeToString(userEddsaPublicKey)
+
+		c.offChainAuthOptionV2 = option.OffChainAuthOptionV2
+		if option.OffChainAuthOptionV2.ShouldRegisterPubKey {
+			for _, sp := range c.storageProviders {
+				registerResult, err := c.RegisterEDDSAPublicKeyV2(sp.EndPoint.Scheme + "://" + sp.EndPoint.Host)
```

Comment:
will there be timeout for calling a single SP endpoint in case SP's pool service affecting the whole functionality? Or do we want to call them concurrently?

Reply:
and might only send to those in_service SP?

Reply:
It does not matter to send those out_of_service SP, because those registered public key could be used when they become in_service.